server.go

```diff
@@ -56,6 +56,12 @@ type ResponseWriter interface {
 	Hijack()
 }
 
+// A ConnectionState interface is used by a DNS Handler to access TLS connection state
+// when available.
+type ConnectionState interface {
```

Comment:
I think idiomatic Go would be for this to be called `ConnectionStater` (see for example [http.Hijacker](https://golang.org/pkg/net/http/#Hijacker) and [http.Pusher](https://golang.org/pkg/net/http/#Pusher)).

Reply:
oh, I thought the original **r** from @miekg 's comment was a typo :s.